### DISCUSSION TITLE
v.2.0beta

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#672] [General]
## WHAT'S CHANGED

BACK-END CHANGES

* Game brought over from Sourceforge to GitHub
* Versioning system converted from CVS to GIT
* Code converted from C to C++
* Code cleanup to make the game easier to modify (#623, #626, #639, #644, $645, among others)
* Parts of the code brought into line with standards (#624)
* Many of the data files and images have been changed to use discrete image files rather than being embedded into the code (#51)
* Vestigial copy protection code finally removed (#493)
* The abbreviation for the sound to play in the Intelligence screens was "INTELLEG" - now "INTELLIG"
* In the config file, "altasLunar" has been corrected to "atlasLunar"

Since transitioning to C++ on GIT, the game has been brought up to where we left off (that is, bugs introduced by the transition have been fixed). Below are improvements that have been made over what the game offered in v.1.1.

BUG FIXES AND PROBLEMS SOLVED

LARGER FIXES
* The duration penalty system has been fixed, and is working right for the first time ever (#246)
    * This makes a final solution to what has sometimes been called the "Duration B Bug"
* The newscast that announces all manned missions will be canceled this turn no longer cancels unmanned missions as well (#164)
* 'Nauts killed in a newscast (train accident, etc.) are now removed immediately (#240)
    * Before, they wouldn't be removed until the start of the following turn, making it possible for the dead crew member to fly on a mission
* If you flew a Docking Duration mission and the Docking step failed, it would cancel the remainder of the mission without attempting Duration (#322)
* If you fly a Lunar Orbital LM Test and the LM Test part fails, the game didn't give you credit for having flown the Lunar Orbital, even though the lunar orbital part was successful. (#332)

SMALLER FIXES
* On a mission where the rocket explodes but the crew escapes thanks to the Launch Escape System pulling the capsule to safety, you still got credit for Manned Space Mission. Now you won't get credit for that unless the mission hits the Reentry step. (#330)
* The newscast announcing that a storm front has blocked a launch window now specifies which pad (#169)
    * Before, it had called it "the H launch facility" (since h is used as a variable).
* When a component faces a Safety penalty unless the player pays to fix it, there was sometimes a mismatch between newscast and actual numbers (#204, #243)
* The VAB/VIB wasn't properly reporting damage to rockets (where you're required to pay xMB or suffer y% Safety loss on its next mission).
* When a component faces a Safety penalty unless the player pays to fix it, it didn't make sure you had enough cash to pay, allowing you to have a negative number of MBs (#268) [not sure if this was in 1.1]
* The second of the two newscasts announcing that the primary crew caught cold and will be replaced with the backup crew never worked before; now it does (#286, #325)
* The newscast stating that an astronaut left to become a test pilot or has been transferred for reeducation instead sent them to the hospital/infirmary (#290)
* The LM failure "Engine explodes upon ignition. Spacecraft crew is killed." used to kill all three crew members instead of just the Commander and LM Pilot (#476)
* If the LM lifts off from the Moon but fails to develop full thrust and makes an emergency landing back on the surface, stranding the crew, the Docking Specialist up in orbit died too (but managed to bring the capsule home anyway) (#287)
* On a lunar landing, "Permanent communications loss with ground control. Mission is scrubbed." would ask who should be first on the Moon, but it shouldn't have because no one was getting out. (#514)
* Because of the nature of the variable for 'naut mood, after a successful mission it could roll over above 100, which would result in a very low number, potentially leading to two entire flight crews resigning out of sudden bitterness during their moment of triumph (#484)
* The newscast that delayed a mission on the pad used to delay missions that couldn't have been put back a turn (such as Mars/Jupiter/Saturn flybys).
* There were three newscasts which were supposed to reduce your budget by 20, 25, or 30 MBs, or by 20%, 25%, or 30% of your budget, whichever is less. Instead, if 20/25/30% was less than 20/25/30 MB, it reduced your budget to 20/25/30% of what it had been, causing your budget to drop like a rock. (#394)
* There was also a newscast that was supposed to increase your budget by 20MB, but the conditions under which it was allowed effectively prevented it ever happening (#397)
* When a newscast gives you three Air Force pilots, they now come with skills. They used to start out in Basic Training with 0s in everything (#541)
* In the Capsule Screen, if you tried to break an empty crew, it would tell you the crew was assigned as the backup crew for a current mission - now it says it's an empty crew (#348)
* Jupiter flybys were set to take 7 turns to arrive, the same as Saturn - now they're set to 4 like they were supposed to be (#171)
* Removed visual artifacts next to astronaut wings and medals in Astronaut/Cosmonaut History (#299, from #192)
* Better centering of astronaut wings in Astro/Cosmonaut history for those who've flown in space.
* Fixed a bug with technology transfer for the Titan/Proton rocket: the technology bonus from multiple rockets at 75%+ was not being applied (#310)
* Fixed a bug in Future Missions: if you chose a mission and proceeded to where you select a capsule or crew, then cancel - then press Continue again (either after choosing a different mission or deciding to go ahead with the same one), and you select capsule or crew, when you go back to Future Missions you find the mission canceled (#222)
* Fixed a bug in the VAB/VIB: with some payloads, the color of the rocket/booster combination wasn't changing to reflect whether it could lift the currently selected payload (#298)
    * This was noticed with an XMS-2 Lunar Pass using B/Titan and a B-Kicker. When you cycled to B/TITAN, the Current Payload displayed in dark red to show that a boosted Titan couldn't lift that combo, but "B/TITAN" under Rocket still showed in white, sometimes leading the player to purchase hardware that couldn't do the job.
* Fixed an array overflow error that used to occasionally crash toward the end of the game (#40, #200)
* Fixed a bug that caused the game to crash with a black screen, if the AI can't find an available crew to send on the mission it's about to fly. This happened most often when it was flying direct ascent. (#552)
* When you're flying docking tests, the game used to get stuck sometimes at 2 DMs in your inventory. For the rest of the game you always had two docking modules and didn't have to buy them again (#319) [not sure if this was in 1.1]
* An "A-OK" failure on an LEM Thrust Test (that is, one that let the test continue successfully) used to not award you LM Point(s) for the test (#441)
* The game used to allow you to assign Duration missions with an illegally low duration (#257, #278)
* When the capsule's emergency-eject system has to be used, the mission would often continue right after, or say it was used successfully but all the crew shows as dead (#244)
* Fixed a bug where a rocket suffers a major fire on the pad, destroying it; the crew survives, but without any mention of the emergency-eject system being used (#547)
* The image for a damaged/destroyed Soviet launch pad instead showed a functioning pad with a rocket ready for launch. It now shows a damaged pad. The damaged-pad icon for the US wasn't too obviously damaged either, so now it's more so. (#412)
* BARIS allowed you to use PgUp and PgDn to scroll through the newscast more efficiently, both at the start of your turn and later in the Viewing Stand. That functionality was lost in RIS because DOS has a peculiar way of processing those keys, but it's now been restored - along with the Home and End keys.
* Mission injuries that led to immediate retirement would retire someone without telling you. Now the retirement is announced in the newscast. (#568)
* Resolved excessive astronaut/cosmonaut retirement checks to match original intent (#428)
* Resolved inconsistencies between the code and the comments when the game checks to see if people should retire.
* In recruitment, if you clicked and held on the up or down arrow, it would take 3 or 4 seconds before the list would start to scroll.
* Addressed an oversight: in recruitment, if you dismissed a candidate, the number of available spots didn't show any higher on the right. (#581)
* The number of spots to fill in recruitment now shows immediately instead of after you've selected to recruit one.
* In recruitment, the Continue button didn't look quite right when clicked on (off by 1px).
* In Preferences, the Human/AI button nestled into the world map never looked quite right on the Soviet side. Now it does. (#353)
* There were two bugs that would cause the game to do a core dump and close if your LM made a fiery impact on the lunar surface. One of them has been fixed (#501)
* When a save is loaded in the Fall, the game would create an extra new intelligence briefing. This could potentially have created an array overflow and overwriting of variables in games that were saved multiple times (#217)
* If your lunar landing failed with the message "Maneuvering failure, craft lands in rough terrain causing damage", the mission would go straight to Lunar Orbital Activities, skipping the liftoff (#505)
* On the lunar landing mission, the Photo Recon check was rolling against the Safety of the wrong device (#482)
* Director rankings were being calculated improperly (#580)

GAME CHANGES

LARGER CHANGES
* Since the duration penalty system now works properly, we've made it possible to downgrade to a lower Duration level in Mission Control (#255)
    * e.g., if you have a Duration C ready to go and schedule a Duration D for next turn, but something goes wrong and the flight doesn't accomplish Duration C, next turn you can downgrade the D mission to C to avoid the -5 penalty for skipping a level.
* We have added back Play By Email. It worked in the CD version of BARIS, but was removed from RIS. (#131) [However, it has been disabled by default.]
* If you have no missions planned for next turn when you hit the Flagpole, you now receive a warning about it. (#166)
* When you open R&D, you will receive a warning if (#166):
    * You've prototyped an LM or Kicker-C but haven't started docking modules yet
    * You've prototyped a landing craft (LM, Kicker-C, direct ascent) but haven't started EVA suits yet
    * You've prototyped Gemini/Voskhod or a minishuttle, but haven't started a kicker yet
* If your cash is at or below 60MB and you have mission(s) scheduled for this turn but haven't assembled them yet, opening R&D will now pop up a reminder to assemble the missions before spending your money on research (#511)
    * If you haven't prototyped a spacecraft yet (ie, you're still on just the satellite), the threshold is 20MB.
* The VAB/VIB now has a Delay button to let you put a launch back a turn rather than having to scrub it and reschedule. (#167)
* In R&D and Purchasing, clicking on the image of the currently displayed hardware item will now pop up a help message giving information about it (#467)
* Similarly, in Future Missions, the name box is now a button that brings up a description of the currently selected mission. (#492)
* In Future Missions, the part of the screen that shows the safety risk is now a button that will launch a pop-up breaking down your requirement penalty into milestone, duration, and new mission penalties. It also shows penalties on the LM steps of lunar landings. (#168, #373)
* In Future Missions, when you select crews, now skills are highlighted only when they'll be used on the mission. LM lights up only on missions that use a lunar module, for instance.
    -In addition, LM Tests without EVA now highlight EV, but less brightly than other skills, because an emergency EVA may be necessary if docking the LM back to the capsule fails.
* When you assign crews to a mission, their members' Endurance is now highlighted along with the skills related to their roles in the crew - since Endurance now helps out on Duration missions. They will not highlight if use of endurance is turned off or game is set to Classic mode. (#586)
* When you select a mission and are at the window to select the type of spacecraft, you can cancel out of there by pressing Esc. Previously you had to hit the Cancel button, but now you can do either.
* Mission Control now shows a figure for requirement penalties: the total of any milestone penalties, duration penalties, and new mission penalty, if any (same as in Future Missions). This way, if say your mission has a milestone or duration penalty you can see it there, and can try downgrading it to remove the penalty. (#320)
* The line for requirement penalties is now color-coded green/yellow/red to match what Future Missions would show in the A-OK!/CAUTION!/DANGER! section.
* Clicking on the image of the pad in Mission Control will now pop up a window that breaks down those requirement penalties. (#168)

SMALLER CHANGES
* Lots of improvements to the help file: grammar corrections, readability, more clarifications, and in a few places updates to reflect additions we've made to the game.
* In single-player games, the Purchasing screen now includes a cheat: pressing $ will add 100MB to your cash on hand. This can be disabled in the config file, and is a Classic setting. (#597)
* When the AI wins the game, you now see its Moon landing play out like other prestige firsts.
* Increased height of Prestige First window so one more prestige first can be listed - which also better centers the window on the screen.
* The game now distinguishes between a damaged and a destroyed pad. Previously, the Launch Facility window showed "DESTROYED" while other windows showed it as "DAMAGED" - regardless what the game had said happened to the pad. (#426)
* It also shows a different image for damaged versus destroyed launch pads. (#466)
* Damaged pads are always 10 or 15MB to repair. Destroyed pads cost 20 or 25MB to fix. (#412)
* The mission name on the pad is now moved slightly to the left - some mission types were crowding the right margin (#270)
* The newscast, and the help pop-up when you press F1, now highlight the up and down arrows when there is more text to scroll up or down to. (#372)
* The arrow keys in the Time Capsule also highlight if you can go up/down
* The arrow keys in the Library at the CIA/KGB also highlight if you can go up/down
* The R&D building now greys out research teams you can't afford. Also, if you are within 5 points of Max R&D, teams will be greyed out so you can't assign more teams than there are points of Safety you can gain by research (#318)
* Saved games are now stored on disk with the name you give them, rather than as BUZZ1.SAV, BUZZ2.SAV, etc.
* In the Capsule Screen, crews assigned to a mission now show in a different color. Those assigned to a mission for this turn show in blue, those for next turn in green; primary crews show in a darker shade. And next to the capsule pic it says that they're Primary or Backup crew for this turn or next. (#173)
* In the Capsule Screen, you can now bring up the Compatibility pop-up by clicking on the astronaut's name - you don't have to click exactly on the morale box.
* To keep track of how much time each person has spent in space, the game used to automatically assign the maximum for each flight's duration level (A=2 days, B=5, C=7, D=12, E=16, F=20), rather than choosing a number of days within each range. As a result, your fellow on the suborbital (a 15-minute flight!) showed as having spent two days in space. Now it assigns a number depending on the type of mission flown (or a random number of days in the range, for E & F). (#300)
* The game now makes some use of the Endurance skill (#384)
    * It contributes to the Duration steps of Duration flights
    * When the newscast randomly selects a spaceperson to be injured, the higher their Endurance the more likely they are to go to be injured rather than simply retire.
* When Mission Control recommends you scrub a mission, if you hit the button to do so, you're now prompted for confirmation so you won't cancel it accidentally, plus it shows a bit more prominently so you're less likely to miss it. (#475)
    * If you should opt to continue the mission, the hardware will suffer a temporary Safety penalty. So this screen now mentions that fact above the button for scrubbing it. (#589)
* When you click on Recruitment and it asks to confirm whether to recruit spacepeople, the game now shows how much money you have to help you decide whether to go ahead with it.
* You can now cancel out of astronaut/cosmonaut recruitment if you change your mind (#295)
* Similarly, the Esc key now cancels you out of the Time Capsule, same as pressing PLAY.
* Recruitment now shows more candidate skills. In addition to Capsule and Endurance, Docking also shows at Astronaut Level 2, and at Level 1, all skills show. This can be returned to the way it used to be in Advanced Preferences. (#176)
* There's now an option in Advanced Preferences that will let you recruit female 'nauts without having to receive the newscast which requires you to recruit them (#329)
* Another option in Advanced Preferences is the Classic setting, which makes the game behave more like it did in original BARIS: you're required to assign backup crews, etc.
* The Max R&D figures for Mercury and Gemini have been reset to their levels in the floppy version (#216)
* Player now has more mission downgrade options - in particular, it is now possible to downgrade a Lunar Landing to a Lunar Orbital LM Test
* If you get the newscast saying you won't be able to purchase rockets for a year, or the equivalent one for capsules, Future Missions will now warn you so you don't accidentally schedule missions you'd just have to cancel next turn. (#611)
* Future Missions now shows the Duration Level (B-F) of manned missions above level A (#251)
* In Mission Control, your downgrade options now include an option to scrub the flight.
* If you scheduled an (Orbit) docking mission (which uses a DM left in orbit by a previous mission) and the following turn there was no DM in orbit, the game was supposed to tell you that it was downgrading the mission to remove the docking part, but that message didn't exist (#288)
* Any time you schedule an (Orbit) mission, Future Missions will now pop up a warning that you will need to launch a DM this turn or next. (#451)
* The opening screen now defaults to the Historical Model rather than the Basic Model (#221)
* The country insignia at the start of a player's turn is now circular; it used to look oval, kind of squashed (see #51)
* In R&D and Purchasing, the Atlas rocket now looks more like an Atlas rocket (#556)
* In R&D and Purchasing, the Soyuz capsule now looks a bit more like the real thing.
* The Boosted Atlas in the VAB looked short and fat, so it's been slimmed a bit.
* The Soyuz image in R&D and Purchasing now looks more like the real thing.
* The LK lander now looks a bit more like the real thing, in R&D and Purchasing, the LM Program building, the VIB, and the CIA/KGB report, though not in mission movies or the Step Failure screen (#588)
* Images of some rockets have had small improvements, both in the VAB/VIB and the intelligence screen: in particular, the N1 looks more historical, the Saturn and Nova have better-looking fins, and the Atlas doesn't look so phallic.
* In the VAB/VIB and the intelligence screen, the nozzles on Soviet rockets used to look a little ragged or uneven - now they look a bit better.
* The VIB image for the Lapot shuttle was too long, causing an invisible error message (unless you ran the program from the command prompt) saying it can't fit the lunar module - and sure enough, you didn't see it on the righthand side in the VIB. That's been fixed. (#418)
* Question marks now look a little better (see middle of #207)
* In R&D, removed the percent sign from the RESEARCH STRONG: +1%/TEAM line, since the research news events change the dice being rolled (7-sided instead of 6-sided, etc.) rather than rolling 6-sided dice and then applying a plus or minus (#321)
* Small updates to the 'naut roster (removal of duplicates, etc.)
* Small updates to the endgame text to prevent text going off the righthand side
* Small updates to the newscast text (e.g., "cannot deliver any capsules or 4-person capsule/modules" was confusing)
* The Soviet newscast now plays the Soviet newscast music (#214)
    * Starting with the CD version of BARIS, and in RIS through v.1.1, it played the US newscast music
    * The Soviet music improves the timing so that the movie's video better matches the voice of the newscaster
* Mission Control and the Launch Facility screen used to play the music for the other side: Soviet instead of US / US instead of Soviet (#258)
* Clarifications have been added to space tenure in 'naut History and to tenure in Basic and Advanced Training: the number is now followed by "seasons" or "days" respectively (#300)
* The Saturn V has been renamed simply "Saturn" - to include other rockets in its family, such as the Saturn IB (#216)
* The Soviet player's default name has been changed from KOROLOV to KOROLYOV, which is a closer transliteration (#262)
* In the VAB/VIB, if you hit Autopurchase but the rocket combo won't lift the current payload, the game now warns you so you don't accidentally buy the wrong hardware. (#516)
* The VAB (but not the VIB) showed a weird stripe up the middle of the rocket fairing, which has been removed. (#515)
* Some hardware (the US B-Kicker, the Lapot, and Soyuz) were showing odd black spots caused by transparency issues in the color palette (#513)
* In Future Missions, the cash figure is now centered under "CASH:"
* Better centering of program names containing "I" (Gemini, Jupiter) in Astronaut Complex / Cosmonaut Center and Future Missions (#345, #361)
* Added Home and End key functionality to Basic and Advanced Training (#508)
* Added Home, End, PgUp, and PgDn functionality to the Prestige Summary and Mission Records screens (#351)
* Added Home, End, PgUp, and PgDn functionality to the Help pop-up (#468)
* Added PgUp, PgDn, Home, and End key functionality in the recruitment screen - especially helpful when female candidates are available (#206)
* Added Home, End, PgUp, and PgDn functionality to the Time Capsule
* Added PgUp, PgDn, Home, and End key functionality to the CIA/KGB Library
* In recruitment, the Continue button is disabled until you've selected a full complement of 'nauts. Now it also pops up a message telling you so. (#462)
* In Hardware Efficiency (off the Museum), the flag in the top left has always been a hidden button linking it to the Prestige Points screen (and vice versa). The flag is now decorated and animated as a button in those screens so as to stand out better.
* In Preferences, the movies/stills option has been disabled (#254)
* When you assign crews to a mission, if you don't have any available crews, the game still popped up a message saying you don't have enough crews to assign a primary and a backup. The old game used to require a backup crew, and the message reflected that. Now it will still show that message if you set the game to require backups, but if not it will say you don't have any crews available. (#381)
* Credits now shows a screen for those who've worked on RIS, in addition to those who worked on the original BARIS (#263)
* The Credits are now in a different color to be easier to read against the background (#211)
* Added names for people who were mentioned in a text file on the CD version but didn't actually make it into the credits for that version
* The result text in the Step Failure screen ("Catastrophic Failure", "All Systems Are Go", etc.) is now centered in its box. It used to be left-justified.
* The Step Failure screen used to underline the crew member who failed a roll, which wasn't very self-explanatory. Now it shows their name in parentheses next to the dice roll results (#474)
* The Step Failure screen used to show the dice roll as Safety VS Roll (e.g., 85 VS 92), which seemed backward. Now it shows Roll vs. Safety (e.g., 92 VS. 85).
* At the end of a single-player game, if you clicked on STATS, it showed your opponent's name as COMPUTER# (e.g., COMPUTER2), the # representing which strategy the computer player used. Now it gives the name of the AI player and shows the number of its (randomly) chosen strategy. (#323)
* If any of the crew on the Moon landing is female, they now show in a different color in the History section of the endgame.
* In the screen that asks who will be first on the Moon, the names have been centered in their respective buttons. Longer names were bleeding out past the edge of the button. (#535)

* NOTE: this list is modified from the release notes as of 9/15/2022.  Items which were implemented after this prerelease (2/17/2022) should all have been removed from this list, but one or two may have been missed. If you find any missing items, please contact Leon at lordpeyre@yahoo.com.

<hr /><em>This discussion was created from the release <a href='https://github.com/raceintospace/raceintospace/releases/tag/v.2.0beta'>v.2.0beta</a>.</em>

**Comment by zombiedesigner** [comment]
Really fantastic results! Well done! 
-Fritz

**Comment by peyre** [comment]
That you, Fritz B.?  Thanks a bunch!  I'm hoping the programmers can pull together a real release soon.

**Reply by zombiedesigner** [reply]
Copy that! Amazing fix list, can’t wait to fly it. 
Really xcited! 

please mention my boardgame sequel in the works www.Liftoff2.com

**Reply by peyre** [reply]
Thanks!  It's good to have you on board - even better to hear you like what you've seen so far.

I'll check out Liftoff 2 - the Docking Module may need some updating re. the game.

**Comment by zombiedesigner** [comment]
Copy that, 
The new version has a lot of revisions and new data,  set up to accommodate for further expansion plans.  
F

**Reply by peyre** [reply]
FYI, we're gearing up to do the release, finally!  Keep an eye out.

**Comment by zombiedesigner** [comment]
Great! When and where? I may have an exciting update to announce soon! Fritz Sent from my iPhoneOn Mar 24, 2025, at 4:37 PM, Leon Baradat ***@***.***> wrote:﻿
FYI, we're gearing up to do the release, finally!  Keep an eye out.

**Reply by peyre** [reply]
Should be on GitHub here, under Releases on the right.   This is the exact URL, I think: https://github.com/raceintospace/raceintospace/releases/tag/untagged-73588eea02460c50651d 

I'll also announce it on my Facebook account and the project's Mastodon account (which has all of 11 followers).

**Comment by zombiedesigner** [comment]
Thx! Greatly looking fwdThe link did not connect.. yet. I will let folks know when it’s announced. when Liftoff! 2.0 goes on Kickstarter please share  the news. Sent from my iPhoneOn Mar 24, 2025, at 11:22 PM, Leon Baradat ***@***.***> wrote:﻿
Should be on GitHub here, under Releases on the right.   This is the exact URL, I think: https://github.com/raceintospace/raceintospace/releases/tag/untagged-73588eea02460c50651d
I'll also announce it on my Facebook account and the project's Mastodon account (which has all of 11 followers).

**Reply by peyre** [reply]
The release is still in draft, unpublished - so you might need to be a member of the team to see it.

**Comment by zombiedesigner** [comment]
Ok no problem Let me know when it’s open Fritz Sent from my iPhoneOn Mar 25, 2025, at 12:51 PM, Leon Baradat ***@***.***> wrote:﻿
The release is still in draft, unpublished - so you might need to be a member of the team to see it.

**Reply by peyre** [reply]
Ok, it's just been released!  [https://github.com/raceintospace/raceintospace/releases/tag/v2.0.0](url)

**Comment by zombiedesigner** [comment]
The list of bug fixing and tweaks and items that were supposed to and then in later versions didn’t occur during the original,  is most impressive and exciting! So many of these items have been on a long wish list of mine and so many new wonderful tweaks that were much needed, Kudos to all involved! Can’t went to play it! I have posted the news and link on the LIFTOFF! 2.0 FB page and will also X.Fritz Fritz Sent from my iPhoneOn Mar 27, 2025, at 1:58 PM, Leon Baradat ***@***.***> wrote:﻿
Ok, it's just been released!  https://github.com/raceintospace/raceintospace/releases/tag/v2.0.0

**Reply by peyre** [reply]
Thanks so much Fritz!  We can use the publicity.

**Reply by peyre** [reply]
P.S. If you have other items on your wish list that aren't in our Issues, we'd be interested to know what they are.

**Comment by zombiedesigner** [comment]
Hi Long ago… what used to function on the “Future Missions” page:   The little button on the left, between the up and down arrow, it used to default to showing all the mission steps for each mission when clicking thru. When the mission steps a visible (or activated) , you should be able to see all the missions with mission bubbles when cycling thru the arrows, up or down. It was curtailed due to a last minute memory issue or memory leak. (That loss really bugged me)Many players do NOT even know this feature of showing mission steps even exists.   At least have the arrow or the little white pixels on the little button cycle and flash to draw attention. I have a few more, but I will have to go thru the game for better recall. FritzSent from my iPhoneOn Apr 8, 2025, at 4:35 pm, Leon Baradat ***@***.***> wrote:﻿
P.S. If you have other items on your wish list that aren't in our Issues, we'd be interested to know what they are.

**Reply by peyre** [reply]
Thanks!  I'll add an Issue for that.

**Reply by peyre** [reply]
So you're saying that if the button is activated, it should show the step bubbles for other missions as you cycle through them - so you don't have to keep pressing it?  And should it be active by default?

**Comment by zombiedesigner** [comment]
Yes, Hitting the button (activating it) would have all missions display the mission steps.  I suggest having the button flash occasionally on a cycle so folks will notice it.  To try it. Sent from my iPhoneOn Apr 8, 2025, at 5:36 pm, Leon Baradat ***@***.***> wrote:﻿
So you're saying that if the button is activated, it should show the mission bubbles for other missions as you cycle through them - so you don't have to keep pressing it?  And should it be active by default?

**Comment by zombiedesigner** [comment]
Prt 2I prefer the default “on”, because it shows the complexity of the mission or simplicity. Sent from my iPhoneOn Apr 8, 2025, at 5:36 pm, Leon Baradat ***@***.***> wrote:﻿
So you're saying that if the button is activated, it should show the mission bubbles for other missions as you cycle through them - so you don't have to keep pressing it?  And should it be active by default?

**Reply by peyre** [reply]
All right, I've added it: [https://github.com/raceintospace/raceintospace/issues/956](url)

**Comment by peyre** [comment]
Say, while I have you, a question's come up.  When the newscast says research will be stronger or weaker for a while, the game rolls a 5, 6, or 7 sided die rather than adding/subtracting from d6 die rolls.  My assumption has always been that the latter was easier to do in a board game, while the computer game took advantage of the computer's easy ability to roll a number from, say, 1 to 7.  Is that correct; was it done deliberately?  Someone here is suggesting that might have been a mistake/bug.

https://github.com/raceintospace/raceintospace/issues/955

**Comment by zombiedesigner** [comment]
That looks like a programmer’s solution to a 1d6 with a, ( + or -) modifier for 1 or the “ rare” 2 modifier. That should be hyper rare btw.    Using the 6, 7 or 8 seems to have the same outcome I assume as the 1d6 modifier strategy? Back then. The programming for true randomness was sometimes a bit off imho at times. I don’t use 2 modifier for 1d6 R&D in liftoff or the new Liftoff 2.0.   That’s a bit of a swing. Can you tell me the odds for it to occur?  Sent from my iPhoneOn Apr 9, 2025, at 2:03 pm, Leon Baradat ***@***.***> wrote:﻿
Say, while I have you, a question's come up.  When the newscast says research will be stronger or weaker for a while, the game rolls a 5, 6, or 7 sided die rather than adding/subtracting from d6 die rolls.  My assumption has always been that the latter was easier to do in a board game, while the computer game took advantage of the computer's easy ability to roll a number from, say, 1 to 7.  Is that correct; was it done deliberately?  Someone here is suggesting that might have been a mistake/bug.
#955

**Reply by peyre** [reply]
Yes, personally I'm sure it's deliberate because a -2 would mean a roll of 1, 2, or 3 would be a 1, and you could have a whole 5 teams' worth of research give you 5-7 points.  You're right that the -2 modifier is rare: there are maybe 3 or 4 newscasts that give you -1, but only one that gives -2.

**Comment by zombiedesigner** [comment]
Good. I would have had  made the 2 as a rare outlier.I wrote the entire game math/data as pseudo-code  (if statements) I had a massive AI design flowchart  for the computer faction that worked beautifully… for months.  There were numerous decision and verification points for choices on reacting to the human player and making it more aggressive or cautious. Also numerous ways the computer player would decide its lunar approach based on lots of circumstances. It had variety and it was quite challenging to play against. Then two weeks before we delivered, Morgan changed something and it was broke. The only solution to deliver in time was to give it more money… it had to cheat, which really bugged me. 2.0 has more scenarios post lunar landing and more ways getting to the moon. 🌙 Play is mostly simultaneous. So it moves fast .  Lots of new tweaks and nuances. Have you seen the weird anim Tom did of an astronaut getting sucked out of the spacecraft? It’s primitive but funny. I think it is an orbital activities roll.  It’s a rare Easter egg. Playing the game likely a thousand times, it never came up for me.   I think I have heard of one player seeing it. Once!   On R&D recently… while testing LIFTOFF 2.0 I have rolled five dice and got sixes on all of them and had a plus one on R&D, getting an increased of 36% improvement. It has happened twice in the last year! I have also had a ton of rolls of 5 dice totalling under  7.  That’s my usual luck.   I now explain the 1d6s dice, representing contractor teams.   (There are good contractor teams and there are bad contractor teams) Sent from my iPhoneOn Apr 9, 2025, at 7:22 pm, Leon Baradat ***@***.***> wrote:﻿
Yes, personally I'm sure it's deliberate because a -2 would mean a roll of 1, 2, or 3 would be a 1, and you could have a whole 5 teams' worth of research give you 5-7 points.  You're right that the -2 modifier is rare: there are maybe 3 or 4 newscasts that give you -1, but only one that gives -2.

**Reply by peyre** [reply]
Hey thanks for the behind-the-scenes info on how we ended up with the whole cheating thing; I've passed it on.  I don't suppose you still have any of those decision & verification flowcharts(?) still around?

No, I haven't seen that animation.  Sounds like a nifty easter egg; (BA)RIS has a few of its own of course, such as the Kicker-C falling over on the Moon's surface.

**Comment by zombiedesigner** [comment]
I have a pile of paperwork and will see if I can rummage thru it to see what is there.  (I try to keep everything)Just been focus’d on Liftoff! 2.0 right now and the contract negotiations with a publisher. Sent from my iPhoneOn Apr 9, 2025, at 7:58 pm, Leon Baradat ***@***.***> wrote:﻿
Hey thanks for the behind-the-scenes info on how we ended up with the whole cheating thing; I've passed it on.  I don't suppose you still have any of those decision & verification flowcharts(?) still around?
No, I haven't seen that animation.  Sounds like a nifty easter egg; (BA)RIS has a few of its own of course, such as the Kicker-C falling over on the Moon's surface.
